### PR TITLE
KAFKA-17421: Add IT for ConsumerRecord#leaderEpoch

### PR DESCRIPTION
**Pull request by chiacyu**
This pr is based on [KAFKA-17421](https://issues.apache.org/jira/browse/KAFKA-17421), please take a look and leave comments. Thanks!

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/clients/consumer/ConsumerRecordTest.java [hunk 1]
```diff
@@ -49,6 +50,7 @@ public void testShortConstructor() {
         assertEquals(ConsumerRecord.NULL_SIZE, record.serializedValueSize());
         assertEquals(Optional.empty(), record.leaderEpoch());
         assertEquals(Optional.empty(), record.deliveryCount());
+        assertNotNull(record.leaderEpoch());
```
Comment (m1a2st) [inline clients/src/test/java/org/apache/kafka/clients/consumer/ConsumerRecordTest.java]: `leaderEpoch` is `Optional<Integer>`, I think the `assertNotnull` isn't a good assertion in here, and `record.leaderEpoch()` already check at line 51.

File: clients/src/test/java/org/apache/kafka/clients/consumer/ConsumerRecordTest.java [hunk 2]
```diff
@@ -98,4 +100,32 @@ record = new ConsumerRecord<>(topic, partition, offset, timestamp, timestampType
...
+        int serializedValueSize = 1142;
+        Optional<Integer> leaderEpoch = Optional.of(1);
+
+        RecordHeaders headers = new RecordHeaders();
+        headers.add(new RecordHeader("header key", "header value".getBytes(StandardCharsets.UTF_8)));
+        ConsumerRecord<String, String> record = new ConsumerRecord<>(topic, partition, offset, timestamp, timestampType,
+                serializedKeySize, serializedValueSize, key, value, headers, leaderEpoch);
+
+        assertEquals(record.leaderEpoch(), leaderEpoch);
+        assertNotNull(record.leaderEpoch());
```
Comment (m1a2st) [inline clients/src/test/java/org/apache/kafka/clients/consumer/ConsumerRecordTest.java]: ditto

**Review by m1a2st — COMMENTED** [review]
Hello @chiacyu, I left some comments, PTAL

**Comment by chiacyu** [comment]
Hi, @m1a2st 
Addressed previously mentioned points, thanks.

**Comment by chia7712** [comment]
@chiacyu thanks for this patch, but what we need is integration test rather than unit test

**Pull request closed without merging**